src/app/leaves/add/components/LeaveRequestFormBuilder.tsx

```diff
@@ -76,6 +81,13 @@ const loginUser: User = {
 };
 
 export const LeaveRequestFormBuilder = () => {
+  const { id: paramsId } = useParams<{ id: string }>();
+  const router = useRouter();
+  const [dateRange, setDateRange] = React.useState<DateRange | undefined>({
```

Comment:
Why do we need this? State in react hook form should be enough.

Reply:
Sir, I have resolved this by adjusting the schema to align with the date range value. Now, we are handling it using a React hook.